mautic/mautic
# TPROD-292: Large refactoring for Symfony 5

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:



#### Steps to test this PR:
Based on https://mautic.atlassian.net/browse/TPROD-292 I noticed rector rules for refactor for Symfony 4 https://github.com/rectorphp/rector/pull/1403/files#diff-e49a81cbb471de97c8ed089b7039c166fd37dcaf315f34652923c107aa0ccaaaR6

I run these rules along with rules from https://mautic.atlassian.net/browse/TPROD-292

This is results for the setup and few manually changes

```
$services->set(\Rector\Renaming\Rector\Name\RenameClassRector::class)
        ->configure([
            'Symfony\Component\HttpKernel\Event\FilterControllerArgumentsEvent'      => 'Symfony\Component\HttpKernel\Event\ControllerArgumentsEvent',
            'Symfony\Component\HttpKernel\Event\FilterControllerEvent'               => 'Symfony\Component\HttpKernel\Event\ControllerArgumentsEvent',
            'Symfony\Component\HttpKernel\Event\FilterResponseEvent'                 => 'Symfony\Component\HttpKernel\Event\ResponseEvent',
            'Symfony\Component\HttpKernel\Event\GetResponseEvent'                    => 'Symfony\Component\HttpKernel\Event\RequestEvent',
            'Symfony\Component\HttpKernel\Event\GetResponseForControllerResultEvent' => 'Symfony\Component\HttpKernel\Event\ViewEvent',
            'Symfony\Component\HttpKernel\Event\GetResponseForExceptionEvent'        => 'Symfony\Component\HttpKernel\Event\ExceptionEvent',
            'Symfony\Component\HttpKernel\Event\PostResponseEvent'                   => 'Symfony\Component\HttpKernel\Event\TerminateEvent',
            ]);
```

After merge we need take  a look on (I revert it back)

In AuthenticationListener.php line 96:
                                                                                                                                                                          
```
  Compile Error: Declaration of Mautic\UserBundle\Security\Firewall\AuthenticationListener::handle(Symfony\Component\HttpKernel\Event\RequestEvent $event) must be comp   
  atible with Symfony\Component\Security\Http\Firewall\ListenerInterface::handle(Symfony\Component\HttpKernel\Event\GetResponseEvent $event)
```




1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Just code review 



<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/11139"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
